__init__.py

```diff
@@ -47,44 +53,50 @@ def handle_go_to_sleep(self, message):
             self.speak_dialog('going.to.sleep', {'wake_word': self.wake_word})
         else:
             self.speak_dialog('going.to.sleep.short')
+        self.perform_sleep()
 
-        self.bus.emit(Message('recognizer_loop:sleep'))
-        self.sleeping = True
-        self.started_by_skill = True
-        wait_while_speaking()
-        time.sleep(2)
-        wait_while_speaking()
-
-        # Dim and look downward to 'go to sleep'
-        # TODO: Get current brightness from somewhere
-        self.old_brightness = 30
-        for i in range(0, (self.old_brightness - 10) // 2):
-            self.enclosure.eyes_brightness(self.old_brightness - i * 2)
-            time.sleep(0.15)
-        self.enclosure.eyes_look("d")
-        platform = self.config_core.get("enclosure").get("platform", "unknown")
-        if platform != "unknown":
-            self.bus.emit(Message('mycroft.volume.mute',
-                                  data={"speak_message": False}))
-        elif self.config_core['confirm_listening']:
-            self.disable_confirm_listening()
-
-    def handle_awoken(self, message):
+    def handle_awoken(self, _) -> None:
         """Handler for the mycroft.awoken message
 
         The message is sent when the listener hears 'Hey Mycroft, Wake Up',
         this handles the user interaction upon wake up.
         """
         started_by_skill = self.started_by_skill
 
-        self.awaken()
+        self.perform_awaken()
         if started_by_skill:
-            self.wake_up_animation()
             # Announce that the unit is awake
             self.speak_dialog("i.am.awake")
-            wait_while_speaking()
+            self.display_waking_face()
+
+    def display_sleep_face(self) -> None:
+        """Display the sleeping face depending on the platform."""
+        if self.gui.connected:
+            self.gui.show_page("resting.qml", override_idle=True)
+        elif self.platform == "mycroft_mark_1":
+            self.display_sleep_animation()
+
+    def display_sleep_animation(self) -> None:
+        """Dim and look downward to 'go to sleep'."""
+        # TODO: Get current brightness from somewhere
+        self.old_brightness = 30
+        for i in range(0, (self.old_brightness - 10) // 2):
+            self.enclosure.eyes_brightness(self.old_brightness - i * 2)
+            time.sleep(0.15)
+        self.enclosure.eyes_look("d")
 
-    def wake_up_animation(self):
+    def display_waking_face(self) -> None:
+        """Display the waking face depending on the platform."""
+        if self.gui.connected:
+            self.gui.remove_page("resting.qml")
+            self.gui.show_page("awake.qml", override_idle=5)
+            # TODO this shouldn't be necessary - remove 2 lines when fixed.
```

Comment:
could you add more information here?  what is the issue?  what is the fix?

Reply:
Improved the comment, but also a good reminder.

@AIIX any idea why an `override_idle=5` wouldn't stop showing the page after 5 seconds?

Reply:
I know what causes it, I'm not sure why that is implemented this way tho, @forslund can maybe tell, Comment out the condition on L427 to L429 till the try exception in mark-2 skill https://github.com/MycroftAI/skill-mark-2/blob/a39ce4d8f6b4469bf6e61e55ae314bf42bf1bbfb/__init__.py#L427

Reply:
Thanks, I've flagged it in that repo and added it to the Skills Interaction sprint.